File: CreationalPatterns/Prototype/Java/Implementations/FarmhousePizza.java
package Implementations;

import Interfaces.AbstractPizza;
import Prototype.IPizzaPrototype;

public class FarmhousePizza extends AbstractPizza {

    public FarmhousePizza(){
        super();
        this.dough = "Fresh Pan";
        this.sauce = "Tomato";
        this.toppings.add("Mushroom");
        this.toppings.add("Capsicum");
        this.toppings.add("Onion");
        this.toppings.add("Tomato");
    }

    public FarmhousePizza(FarmhousePizza source){
        super(source);
    }

    @Override
    public IPizzaPrototype copy() {
        return new FarmhousePizza(this);
    }    
}

File: CreationalPatterns/Prototype/C#/Prototype/Implementations/FarmhousePizza.cs
public class FarmhousePizza : AbstractPizza {

    public FarmhousePizza(){
        this.Dough = "Fresh Pan";
        this.Sauce = "Tomato";
        this.Toppings.Add("Mushroom");
        this.Toppings.Add("Capsicum");
        this.Toppings.Add("Onion");
        this.Toppings.Add("Tomato");
    }

    public FarmhousePizza(FarmhousePizza source) : base(source){
    }

    public override IPizzaPrototype Copy() {
        return new FarmhousePizza(this);
    }    
}
